### ISSUE TITLE
[Bug] 2 actions with the same name in different categories won't work

### ISSUE THREAD

**Issue by NGumby** [thomasgalliker/Plugin.FirebasePushNotifications#48]
### Description

Because GetNotificationAction doesn't check for the category of the notification, you can't have the same action in 2 different categories

### Steps to Reproduce

            var notificationCategories = new NotificationCategory[]
               {
                    new NotificationCategory("dismiss",new NotificationAction[] {
                        new NotificationAction("Dismiss","Dismiss", NotificationActionType.Default),
                    }),
                    new NotificationCategory("navigate",new NotificationAction[] {
                        new NotificationAction("Dismiss","Dismiss", NotificationActionType.Default),
                        new NotificationAction("Navigate","Navigate To", NotificationActionType.Foreground)
                    })
            };
            CrossFirebasePushNotification.Current.RegisterNotificationCategories(notificationCategories);

send a notification with click_action:"navigate"
click on the Dismiss button

### Expected Behavior
- NotificationAction event triggered with Dismiss button

### Actual Behavior
[DOTNET] 2024-08-10 15:20:13.7510 - CSAPP - ERROR - OnReceive failed with exception
[0:] Plugin.FirebasePushNotifications.Platforms.PushNotificationActionReceiver: Error: OnReceive failed with exception

System.InvalidOperationException: Sequence contains more than one matching element
   at System.Linq.ThrowHelper.ThrowMoreThanOneMatchException()
   at System.Linq.Enumerable.TryGetSingle[NotificationAction](IEnumerable`1 source, Func`2 predicate, Boolean& found)
   at System.Linq.Enumerable.SingleOrDefault[NotificationAction](IEnumerable`1 source, Func`2 predicate)
   at Plugin.FirebasePushNotifications.Platforms.FirebasePushNotificationManagerBase.GetNotificationAction(String notificationActionId)
   at Plugin.FirebasePushNotifications.Platforms.FirebasePushNotificationManagerBase.HandleNotificationAction(IDictionary`2 data, String notificationActionId, NotificationCategoryType notificationCategoryType)
   at Plugin.FirebasePushNotifications.Platforms.PushNotificationActionReceiver.OnReceive(Context context, Intent intent)

### Basic Information

- Version with issue: 2.2.109-pre
- Last known good version:

**Comment by thomasgalliker** [comment]
True. There is some room for improvment. Thanks for reporting 👍🏻

**Comment by thomasgalliker** [comment]
There was no code to ensure that the user tapped the notification action of a certain notification category.
I extended the GetNotificationAction method accordingly and you'll have the fix in release 2.2.115-pre (and newer).

I did use following notification message to test if it works. I did test on an iPhone 8 and on a Google Pixel 5. 
```
{
    "message": {
        "topic": "general",
        "notification": {
            "title": "Navigation",
            "body": "Want start navigation?"
        },
        "data": {
            "priority": "high",
            "click_action": "navigate"
        },
        "apns": {
            "payload": {
                "aps": {
                    "category": "navigate"
                }
            }
        }
    }
}
```
Can you please also do some tests and let me know if it works as expected?
